### PR TITLE
changed signedness of iterator to fix warnings.

### PR DESCRIPTION
**Pull request by ghost**
numOperands is an unsigned integer. In addition, getArgOperand has unsigned int arguments. As a result, I have no idea why i is an int, but to fix compiler warnings, I made it an unsigned int.

### CODE REVIEW

File: src/mono/mono/mini/mini-llvm-cpp.cpp [hunk 1]
```diff
@@ -382,7 +382,7 @@ mono_llvm_call_args (LLVMValueRef wrapped_calli)
 
 	LLVMValueRef *ret = g_malloc (sizeof (LLVMValueRef) * numOperands);
 
-	for (int i=0; i < numOperands; i++) {
+	for (unsigned int i=0; i < numOperands; i++) {
```
Comment (jaykrell) [inline src/mono/mono/mini/mini-llvm-cpp.cpp]: ‘guint’ or ‘size_t’?

**Review by lambdageek — APPROVED** [review]

File: src/mono/mono/mini/mini-llvm-cpp.cpp (hunk 1 above)
Comment (lambdageek) [reply src/mono/mono/mini/mini-llvm-cpp.cpp]: `llvm::CallBase::getNumArgOperands` returns `unsigned`.

**Review by imhameed — APPROVED** [review]

File: src/mono/mono/mini/mini-llvm-cpp.cpp (hunk 1 above)
Comment (imhameed) [reply src/mono/mono/mini/mini-llvm-cpp.cpp]: It's too bad that there isn't a more humane way to write `for (typename std::decay<decltype(numOperands)>::type i = 0; i < numOperands; i++)`.

Comment (jaykrell) [reply src/mono/mono/mini/mini-llvm-cpp.cpp]: “guint” is just a mono abbreviation for same, and “size_t” interoperates losslessly & efficiently in this context, albeit maybe slight overkill.

**Comment by jaykrell** [comment]
Mono is full of signed/unsigned mixups.
I have resisted fixing them because “mono likes int” and in general must be careful.

**Pull request merged**